### PR TITLE
make select star not show hll derived columns

### PR DESCRIPTION
**Pull request by jiaqi-g**
make select star not show hll derived columns

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionFetcher.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
   private final SelectionColumnIterator[] selectionColumnIterators;
 
   public SelectionFetcher(Block[] blocks, DataSchema dataSchema) {
-    this.length = blocks.length;
+    this.length = dataSchema.size();
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionFetcher.java]: how will this work, are we assuming blocks.length == dataschema.size().

Comment (victor2100) [inline pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionFetcher.java]: Yes, this should not be changed. From the logic we assume  blocks.length == dataschema.size(). I updated the code and you can take a look at it.

File: pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java [hunk 2]
```diff
@@ -144,7 +146,15 @@ public static SelectionResults renderSelectionResultsWithoutOrdering(@Nonnull Co
   public static List<String> getSelectionColumns(@Nonnull List<String> selectionColumns,
       @Nonnull IndexSegment indexSegment) {
     if (selectionColumns.size() == 1 && selectionColumns.get(0).equals("*")) {
-      List<String> allColumns = Arrays.asList(indexSegment.getColumnNames());
+      List<String> allColumns = new ArrayList<>();
+      for (String column: indexSegment.getColumnNames()) {
+        FieldSpec spec = indexSegment.getSegmentMetadata().getSchema().getFieldSpecFor(column);
+        if (spec instanceof MetricFieldSpec && ((MetricFieldSpec) spec).isDerivedMetric()) {
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java]: why are we skipping all derived metrics? we need to introduce a new boolean selectVisibility in fieldSpec. When it is set to false (default is true), we can to remove the column from selection result. For autogenerated columns like hll we can set it to false.

Comment (antumbde) [inline pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java]: @kishoreg  For the time-column related examples you brought up, do those columns need to be "derived" ? They can be regular columns in the metadata if we are not storing any association.

Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java]: yes, they will be derived from other columns and we want them to be visible in the selection result.

**Comment by Jackie-Jiang** [comment]
I think we can postpone this change because how to handle the visibility of columns is still under discussion. We should be able to handle this more gracefully.

File: pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java (hunk 2 above)
Comment (antumbde) [inline pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java]: yes, but do they need "derived" annotation in metadata. Are they any different from regular columns ?

**Comment by kishoreg** [comment]
yes, we need that metadata in schema. Its optional in metadata but currently we generate schema from metadata since schema is not enforced. Without that we don't have enough info during segment generation that we have to automatically create that column. Currently, fieldspec is tied with schema and segment metadata

**Comment by victor2100** [comment]
@Jackie-Jiang @kishoreg I added the isSelectStarVisible as a column metadata, it's up to you guys whether to merge this or not.

**Comment by npawar** [comment]
Closed this PR due to 6 months inactivity. Reopen if needed.

**Pull request closed without merging**